Makefile

```diff
@@ -0,0 +1,81 @@
+DOCKER          = docker
+DOCKER_COMPOSE  = docker-compose
+
+##
+## ----------------------------------------------------------------------------
+##   Environment
+## ----------------------------------------------------------------------------
+##
+
+build: ## Build the environment
+	$(DOCKER_COMPOSE) build
+
+install: ## Install the environment
+	make build start composer yarn encore-prod
+	open http://localhost:8000
+
+logs: ## Follow logs generated by all containers
+	$(DOCKER_COMPOSE) logs -f --tail=0
+
+logs-full: ## Follow logs generated by all containers from the containers creation
+	$(DOCKER_COMPOSE) logs -f
+
+ps: ## List all containers managed by the environment
+	$(DOCKER_COMPOSE) ps
+
+restart: ## Restart the environment
+	$(DOCKER_COMPOSE) restart
+
+start: ## Start the environment
+	$(DOCKER_COMPOSE) up -d --remove-orphans
+
+stats: ## Print real-time statistics about containers ressources usage
+	$(DOCKER) stats $($(DOCKER) ps --format={{.Names}})
+
+stop: ## Stop the environment
+	$(DOCKER_COMPOSE) stop
+
+uninstall: ## Uninstall the environment
+	$(DOCKER_COMPOSE) kill
+	$(DOCKER_COMPOSE) down --volumes --remove-orphans
+
+.PHONY: build install logs logs-full ps restart start stats stop uninstall
+
+##
+## ----------------------------------------------------------------------------
+##   Project
+## ----------------------------------------------------------------------------
+##
+
+composer: ## Install Composer dependencies from the "php" container
+	$(DOCKER_COMPOSE) exec php sh -c "composer install --optimize-autoloader"
```

Comment:
From my understanding, there is no real issue with `--optimize-autoloader` as its purpose is to generate a classmap file to avoid filesystem checks. A thing quite interesting in term of performances while you are using OPcache.

If a class is not present in the classmap, the autoload will still request the filesystem to load it. You can find more details a little bit further in the documentation you linked (https://getcomposer.org/doc/articles/autoloader-optimization.md#trade-offs).